totalRP3/Modules/Register/Characters/RegisterIgnore.lua

```diff
@@ -1,120 +1,21 @@
-	local profile = getPlayerCurrentProfile();
-	if not profile.relation then
-		profile.relation = {};
-	end
-	profile.relation[profileID] = relation;
-end
-TRP3_API.register.relation.setRelation = setRelation;
-
-local function getRelation(profileID)
-	local relationTab = get("relation") or EMPTY;
-	return relationTab[profileID] or RELATIONS.NONE;
-end
-TRP3_API.register.relation.getRelation = getRelation;
-
-local function getRelationText(profileID)
-	local relation = getRelation(profileID);
-	if relation == RELATIONS.NONE then
-		return "";
-	end
-	return loc:GetText("REG_RELATION_" .. relation);
-end
-TRP3_API.register.relation.getRelationText = getRelationText;
-
-local function getRelationTooltipText(profileID, profile)
-	return loc:GetText("REG_RELATION_" .. getRelation(profileID) .. "_TT"):format(getPlayerCompleteName(true), getCompleteName(profile.characteristics or EMPTY, UNKNOWN, true));
-end
-TRP3_API.register.relation.getRelationTooltipText = getRelationTooltipText;
-
-local function getRelationTexture(profileID)
-	return RELATIONS_TEXTURES[getRelation(profileID)];
-end
-TRP3_API.register.relation.getRelationTexture = getRelationTexture;
-
-local function getRelationColors(profileID)
-	local relation = getRelation(profileID);
-	if relation == RELATIONS.NONE then
-		return 1, 1, 1;
-	elseif relation == RELATIONS.UNFRIENDLY then
-		return 1, 0, 0;
-	elseif relation == RELATIONS.NEUTRAL then
-		return 0.5, 0.5, 1;
-	elseif relation == RELATIONS.BUSINESS then
-		return 1, 1, 0;
-	elseif relation == RELATIONS.FRIEND then
-		return 0, 1, 0;
-	elseif relation == RELATIONS.LOVE then
-		return 1, 0.5, 1;
-	elseif relation == RELATIONS.FAMILY then
-		return 1, 0.75, 0;
-	end
-end
-TRP3_API.register.relation.getRelationColors = getRelationColors;
-
--- TODO Move this somewhere that makes sense. Also, Saelora should have done this a long time ago :P
-local NEUTRAL = Ellyb.Color.CreateFromRGBA(0.5, 0.5, 1, 1):Freeze()
-local BUSINESS = Ellyb.Color.CreateFromRGBA(1, 1, 0, 1):Freeze()
-local LOVE = Ellyb.ColorManager.PINK;
-local FAMILY = Ellyb.Color.CreateFromRGBA(1, 0.75, 0, 1):Freeze()
-function TRP3_API.register.relation.getColor(relation)
-	if relation == RELATIONS.UNFRIENDLY then
-		return Ellyb.ColorManager.RED;
-	elseif relation == RELATIONS.NEUTRAL then
-		return NEUTRAL;
-	elseif relation == RELATIONS.BUSINESS then
-		return BUSINESS;
-	elseif relation == RELATIONS.FRIEND then
-		return Ellyb.ColorManager.GREEN;
-	elseif relation == RELATIONS.LOVE then
-		return LOVE;
-	elseif relation == RELATIONS.FAMILY then
-		return FAMILY;
-	else
-		return Ellyb.ColorManager.WHITE;
-	end
-end
+local tinsert = tinsert;
+local Relation = TRP3_API.register.relation
+local setRelation = Relation.setRelation
+local getRelationList = Relation.getRelationList
+local getRelationText = Relation.getRelationText
+local getRelationTexture = Relation.getRelationTexture
+local getRelationTooltipText = Relation.getRelationTooltipText
```

Comment:
Generally speaking I'd rather function calls just be fully qualified at each call-site for searchability.

Nothing irks me more than trying to trace our function usage in the addon just to get a wall of search results that lead me to the first ~30 lines of a file where half the things being imported were renamed. In this case I'd mutter very unkind words if I had to grep for `TRP3_API.register.relation.*` two months down the line.

While fully qualifying each call site does add verbosity, _most_ of the problem there is the Java-esque overuse of namespace tables within the addon. That's a much larger topic to deal with however.

As for `tinsert`, it really doesn't need to be upvalued.

Reply:
Agree entirely, i'll switch these back.